source/Halibut/Util/AsyncServiceVerifier.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Halibut.Exceptions;
+
+namespace Halibut.Util
+{
+    public class AsyncServiceVerifier
+    {
+        /// <summary>
+        /// This method enforces that all methods on the sync service have a corresponding method on the async service which adheres to the following conventions:
```

Comment:
Is it intentional that the async interface can have methods not on the TSyncService?

Reply:
It's intentional that we don't care if there are, yes. I figured that you're always looking for an async version of a requested sync service, and the important part is that the async interface should cover the whole sync interface. If the async interface has some extra methods, then so what?

We could be stricter here, but I'm not sure it gives much benefit.